### ISSUE TITLE
Save as template / Custom Template

### ISSUE THREAD

**Issue by jauyong** [GoogleForCreators/web-stories-wp#223]
As an author I want the ability to save a story as a template for future use

**[Feature Brief - Epic: Document Workflow](https://docs.google.com/document/d/1GJZljicdXcP7oRdywsF3EGUaCVpCHsDrK2gUMubnrbk/edit#heading=h.89cwdir5ijn5)**

**UX designs TBD**

**Acceptance Criteria**
1. This functionality is invoked by triggering the “Save as template” option in the Stories list/dashboard more options menu or in the workspace. 
2. When a user triggers “save as template”, we create a copy of the story (the original stays as it is) but as a template which then shows in the saved/custom templates section. The template name is the title of the story used to create the template. 
3. A user can “save as template” from any existing story in their dashboard, either published or draft. 
4. A user is able to modify the template in the editor and save their updates. 
5. In the saved/custom templates section, the user is able to rename a custom template via a "rename" option. 

## Reviews
- [x] Product brief reviewed by UX
- [x] Product brief reviewed by eng (TL and/or dev)
- [ ] UX Design reviewed and approved by PM
- [ ] UX Design reviewed and approved by eng
- [ ] Eng design reviewed by TL (if needed)

**Comment by jauyong** [comment]
@samitron7 has the UX and will move over to Figma
Note: UX also needed for when editing a template

This is a really nice feature to have, but we are also trying to avoid adding new features during Alpha - Beta. Not a top priority amongst other UX needed items right now

**Comment by samitron7** [comment]
We should move this ticket out of beta

**Comment by BrittanyIRL** [comment]
This ticket is unblocked by https://github.com/google/web-stories-wp/pull/1997 which was merged earlier today!

**Comment by swissspidy** [comment]
Some additional work on the PHP side is likely required here, as we've implemented _templates_ as their own content type that sits next to the _story_ content type, for various reasons.

So "Save as template” would mean "switch this entity's type from 'story' to 'template'".

That means the _template_ content type also needs to be able to be edited in the editor, albeit with limited functionality ("the publish button and related areas will disappear").

**Comment by BrittanyIRL** [comment]
Pretty sure this is an echo of what @swissspidy called out on Friday. 

Am able to save as a template but these requirements are not available yet. I think answering some of these in more detail will help Pascal and team be able to flesh out the API a bit more too. 

- After saving as a template, the publish button and related areas will disappear, as you’re now editing an unpublishable template 
  - This, to me, means that this story is now ONLY a template unlike creating a story from a template where the template stays a template. I'm curious is this is necessary? I think it makes sense to create a template based on a story and still allow that story to exist - what happens if it's the first in a campaign of some sort and then you create a template from it and try to publish it - seems like it could be a round about user experience? 
  - If we are meant to "remove publish and related areas" then i think changing the status of the former story to a template would help us block this functionality. 

- After saving as a template, the saved template will not appear in the Dashboard under drafts,, and will instead appear under “My templates” in the templates gallery
  - I think the 'templates gallery' here means the 'saved templates' view, yes? 


@o-fernandez can you help expand on this so we know how to move forward?

**Comment by o-fernandez** [comment]
Since this is not in scope for the Beta, and we don't have bandwidth, I temporarily moved it to the Backlog. But let me know if this helps. @swissspidy @BrittanyIRL 

- I actually think that "save as template" could be a function we only expose in the "my stories" dashboard view. So, in essence, it will not be something the user does from the editor but rather from the dashboard. 
- In this model, the user sees a story they want to save as template, clicks "save as template" and we create a copy of the story (the original stays as it is) which then shows in the custom templates for that user. The user can do this from a published story or a draft/unpublished story. 
- I don't think it's necessary for the user to be able to edit/update a saved template, they could just create a story from a template, make edits, and then create a new saved template from that modified version. 

@samitron7 and others, feel free to chime in if you think it's critical to save a story as template from the editor directly.

**Comment by swissspidy** [comment]
That'd be much easier to implement on the server side, that's for sure 👍

**Comment by BrittanyIRL** [comment]
This sounds like a great plan. Thanks for the feedback. 

I also think for the time being i might disable this action from the my stories view in the dashboard so that people can't click it and make a billion saved templates that they will not be able to see.

**Comment by o-fernandez** [comment]
Updated the ticket description/acceptance criteria with this approach. Let's plan for this and we can adjust in the future, as necessary, based on user feedback.

**Comment by samitron7** [comment]
@o-fernandez I commented on the brief but I think being able to save template inside the editor is a nice (and pretty common practice in design tools) to have at some point in the future. I already have ui for it (dropdown arrow next to title exposes action like save to template, make a copy etc..).

**Comment by o-fernandez** [comment]
@samitron7 updated the brief to include template editing, though that requires additional UX specs for your consideration which I've added to the spec. Will live this in UX needed to finalize the open considerations, let me know what you think.

**Comment by samitron7** [comment]
@o-fernandez Got it. I do have design consideration for this already and will pull the designs into figma when this ticket is up on my queue.
